### PR TITLE
Dance party: Don't allow song to be changed while running

### PR DESCRIPTION
**Pull request by breville**
This disables changing song while the dance party level is running.  

Dance party has a kind of pre-run state in which the user has pressed Run, but it's still loading assets, and so we disable the song selection during that state too.

Fixes https://github.com/code-dot-org/dance-party/issues/190.

### CODE REVIEW

File: apps/src/dance/Dance.js [hunk 1]
```diff
@@ -395,6 +397,7 @@ Dance.prototype.runButtonClick = async function () {
   this.studioApp_.toggleRunReset('reset');
   // Safe to allow normal run/reset behavior now
   this.runIsStarting = false;
+  getStore().dispatch(setRunIsStarting(false));
```
Comment (Bjvanminnen) [inline apps/src/dance/Dance.js]: do we still need `this.runIsStarting`?

**Comment by breville** [comment]
Oh, the description is an extra detail in addition to the title.  I'll update :)

File: apps/src/dance/Dance.js (hunk 1 above)
Comment (breville) [reply apps/src/dance/Dance.js]: Ah, I can use redux now.  Good idea.

**Comment by joshlory** [comment]
+1 for adding a short regression test!

**Comment by breville** [comment]
UI test coverage added.

**Pull request merged**